src/module_wrap.cc

```diff
@@ -90,13 +87,22 @@ void ModuleWrap::New(const FunctionCallbackInfo<Value>& args) {
 
   Local<String> url = args[1].As<String>();
 
+  Local<Object> that = args.This();
+
+  Local<Value> options = args[2];
+  MaybeLocal<Integer> lineOffset = ModuleWrap::GetLineOffsetArg(env, options);
+  MaybeLocal<Integer> columnOffset =
+    ModuleWrap::GetColumnOffsetArg(env, options);
+  MaybeLocal<Context> maybe_context = ModuleWrap::GetContext(env, options);
+
+  Local<Context> context = maybe_context.FromMaybe(that->CreationContext());
   Local<Module> module;
 
   // compile
   {
     ScriptOrigin origin(url,
```

Comment:
Use a `Context::Scope context_scope(context)` here. It may look like `ScriptCompiler` is context-agnostic, but in reality it does depend on the current context. In other words, copy from `ContextifyScript` as much as possible, ignoring the cached data stuff:

https://github.com/nodejs/node/blob/604578f47ea360980110e2cd7d4a636f9942b1f0/src/node_contextify.cc#L652-L677